nomad/job_endpoint.go

```diff
@@ -89,6 +89,24 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 		if !aclObj.AllowNsOp(args.RequestNamespace(), acl.NamespaceCapabilitySubmitJob) {
 			return structs.ErrPermissionDenied
 		}
+		// Validate Volume Permsissions
+		for _, tg := range args.Job.TaskGroups {
+			for _, vol := range tg.Volumes {
+				if vol.Type != structs.VolumeTypeHost {
+					return structs.ErrPermissionDenied
```

Comment:
not a big deal but this seems more invalid request or unsupported type than a permission error, same as the parsing error below.

Reply:
Yep - it'll actually be caught earlier in the process, (validation of the submission happens before permissions validation), this is entirely being overly defensive.